backends/smt2/smtbmc.py

```diff
@@ -969,6 +976,30 @@ def write_vcd_trace(steps_start, steps_stop, index):
 
         vcd.set_time(steps_stop)
 
+def detect_state_loop(steps_start, steps_stop):
+    print_msg(f"Checking for loops in found induction counter example")
+    print_msg(f"This feature is experimental and incomplete")
+
+    path_list = sorted(smt.hiernets(topmod, regs_only=True))
+
+    mem_trace_data = collect_mem_trace_data(steps_start, steps_stop)
+
+    # Map state to index of step when it occurred
+    states = dict()
+
+    for i in range(steps_start, steps_stop):
+        value_list = smt.get_net_bin_list(topmod, path_list, "s%d" % i)
+        mem_state = sorted(
+            [(tuple(path), addr, data)
+             for path, addr, data in mem_trace_data.get(i, [])])
+        state = tuple(value_list), tuple(mem_state)
+        if states.get(state):
```

Comment:
Missed this the first time: When `steps_start` is 0 we can end up with a state at time step 0 which is false in a boolean context and thus a loop with that state would be ignored here. Use `if state in states:` instead.

Reply:
certified 🐍 moment